### PR TITLE
done

### PR DESCRIPTION
**Pull request by AlinaSastras**

### CODE REVIEW

**Review by adrianlibenciuc — DISMISSED** [review]

File: SQLQuery1.sql [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+DROP TABLE IF EXISTS icn_user_role
+CREATE TABLE icn_user_role(
+	id_user integer not null,
+	id_role integer not null,
+	FOREIGN KEY (id_user) REFERENCES icn_user(id),
+	FOREIGN KEY (id_role) REFERENCES icn_role(id),
+	)
+
+DROP TABLE IF EXISTS patient
+CREATE TABLE patient(
```
Comment (baciu-radu) [inline SQLQuery1.sql]: has to have icn_

File: SQLQuery1.sql [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+CREATE TABLE icn_role(
+	id	integer	not null identity primary key,
+	icn_role_name varchar(20),
+	)
+
+DROP TABLE IF EXISTS icn_user;
+CREATE TABLE icn_user(
+	id integer not null identity primary key,
+	email varchar(80),
+	icn_password varchar(20),
```
Comment (baciu-radu) [inline SQLQuery1.sql]: add account_status aswell as int

File: SQLQuery1.sql [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+)
+
+DROP TABLE IF EXISTS icn_user_role
+CREATE TABLE icn_user_role(
+	id_user integer not null,
```
Comment (baciu-radu) [inline SQLQuery1.sql]: PK (id_user, id_role)

File: SQLQuery1.sql [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+	id_role integer not null,
+	FOREIGN KEY (id_user) REFERENCES icn_user(id),
+	FOREIGN KEY (id_role) REFERENCES icn_role(id),
+	)
+
+DROP TABLE IF EXISTS patient
+CREATE TABLE patient(
+	id integer NOT NULL IDENTITY PRIMARY KEY,
+	id_user integer,
```
Comment (baciu-radu) [inline SQLQuery1.sql]: NOT_NULL

File: SQLQuery1.sql [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+	cnp varchar(14),
```
Comment (baciu-radu) [inline SQLQuery1.sql]: NOT_NULL

Comment (baciu-radu) [inline SQLQuery1.sql]: UNIQUE

File: SQLQuery1.sql [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+
+DROP TABLE IF EXISTS specialization
+CREATE TABLE specialization(
+	id integer NOT NULL IDENTITY PRIMARY KEY,
+	s_name varchar(40),
+	)
+
+DROP TABLE IF EXISTS doctor_specialization
+CREATE TABLE doctor_specialization(
+	d_id integer,
```
Comment (baciu-radu) [inline SQLQuery1.sql]: doctor_id, specialization_id

File: SQLQuery1.sql [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+	s_id integer,
+	FOREIGN KEY (s_id) REFERENCES specialization(id),
+	FOREIGN KEY (d_id) REFERENCES doctor(id),
+	)
+
+--DROP TABLE IF EXISTS icn_location
```
Comment (baciu-radu) [inline SQLQuery1.sql]: delete

File: SQLQuery1.sql [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+--CREATE TABLE icn_location(
+--	id integer NOT NULL IDENTITY PRIMARY KEY,
+--	details varchar(200),
+--	)
+
+DROP TABLE IF EXISTS appointment
+CREATE TABLE appointment(
+	id integer NOT NULL IDENTITY PRIMARY KEY,
+	Pid integer,
```
Comment (baciu-radu) [inline SQLQuery1.sql]: patient_id

File: SQLQuery1.sql [hunk 9]
```diff
@@ -0,0 +1,91 @@
...
+	Did integer,
```
Comment (baciu-radu) [inline SQLQuery1.sql]: doctor_id

File: SQLQuery1.sql (hunk 8 above)
Comment (baciu-radu) [inline SQLQuery1.sql]: not_null

File: SQLQuery1.sql [hunk 10]
```diff
@@ -0,0 +1,91 @@
...
+--	Lid integer,
+	Details varchar(200),
+	FOREIGN KEY (Pid) REFERENCES patient(id),
+	FOREIGN KEY (Did) REFERENCES doctor(id),
+--	FOREIGN KEY (Lid) REFERENCES icn_location(id),
```
Comment (baciu-radu) [inline SQLQuery1.sql]: delete

File: SQLQuery1.sql [hunk 11]
```diff
@@ -0,0 +1,91 @@
...
+--	id integer NOT NULL IDENTITY PRIMARY KEY,
+--	details varchar(200),
+--	)
+
+DROP TABLE IF EXISTS appointment
+CREATE TABLE appointment(
+	id integer NOT NULL IDENTITY PRIMARY KEY,
+	Pid integer,
+	Did integer,
+--	Lid integer,
```
Comment (baciu-radu) [inline SQLQuery1.sql]: delete

File: SQLQuery1.sql [hunk 12]
```diff
@@ -0,0 +1,91 @@
...
+--	FOREIGN KEY (Lid) REFERENCES icn_location(id),
+	)
+
+
+
+
+INSERT INTO icn_role VALUES ('administrator'),('doctor'),('patient')
+INSERT INTO icn_user VALUES ('andrei.mladin@gmail.com','pascal'),('radumbaciu@gmail.com','pascal'),('libenciuc.adrian@yahoo.com','pascal'),('florentina.mirisan@yahoo.ro','pascal'),('sastrasalina@gmail.com','pascal')
+INSERT INTO icn_user_role VALUES (1,1),(2,2),(3,3),(4,3),(5,3)
+--INSERT INTO icn_location VALUES ('locatia1'),('locatia2'),('locatia3')
```
Comment (baciu-radu) [inline SQLQuery1.sql]: delete

File: SQLQuery1.sql [hunk 13]
```diff
@@ -0,0 +1,91 @@
...
+INSERT INTO patient VALUES (3,'1234567890123','Libenciuc','Adrian',1),(4,'2345678901123','Florentina','Mirisan',1),(5,'3456789012123','Alina','Sastras',0)
+INSERT INTO doctor VALUES (2,'Radu','Baciu')
+INSERT INTO specialization VALUES ('Allergy and immunology'),(' Anesthesiology'),(' Dermatology'),(' Diagnostic radiology'),(' Emergency medicine'),(' Family medicine'),(' Internal medicine'),(' Medical genetics')
+INSERT INTO doctor_specialization VALUES(1,1),(1,2),(1,3),(1,4)
+--INSERT INTO appointment VALUES(1,1,1,'detalii apointment1'),(2,1,2,'detalii apointment2')
```
Comment (baciu-radu) [inline SQLQuery1.sql]: delete

File: SQLQuery1.sql [hunk 14]
```diff
@@ -0,0 +1,91 @@
...
+INSERT INTO appointment VALUES(1,1,'detalii apointment1'),(2,1,'detalii apointment2')
+
```
Comment (baciu-radu) [inline SQLQuery1.sql]: delete empty line

**Review by baciu-radu — CHANGES_REQUESTED** [review]

**Comment by baciu-radu** [comment]
change User entity >>> change status to account_status
                                >>> change password to pwd

File: SQLQuery1.sql [hunk 15]
```diff
@@ -0,0 +1,91 @@
...
+	Details varchar(200),
+	FOREIGN KEY (Pid) REFERENCES patient(id),
+	FOREIGN KEY (Did) REFERENCES doctor(id),
+--	FOREIGN KEY (Lid) REFERENCES icn_location(id),
+	)
+
+
+
+
+INSERT INTO icn_role VALUES ('administrator'),('doctor'),('patient')
```
Comment (baciu-radu) [inline SQLQuery1.sql]: roles have to be uppercase ADMIN DOCTOR PATIENT

File: SQLQuery1.sql [hunk 16]
```diff
@@ -0,0 +1,91 @@
+CREATE database iCareNow;
```
Comment (baciu-radu) [inline SQLQuery1.sql]: all passwords should be encoded

File: SQLQuery1.sql [hunk 17]
```diff
@@ -0,0 +1,91 @@
...
+	FOREIGN KEY (Pid) REFERENCES patient(id),
+	FOREIGN KEY (Did) REFERENCES doctor(id),
+--	FOREIGN KEY (Lid) REFERENCES icn_location(id),
+	)
+
+
+
+
+INSERT INTO icn_role VALUES ('administrator'),('doctor'),('patient')
+INSERT INTO icn_user VALUES ('andrei.mladin@gmail.com','pascal'),('radumbaciu@gmail.com','pascal'),('libenciuc.adrian@yahoo.com','pascal'),('florentina.mirisan@yahoo.ro','pascal'),('sastrasalina@gmail.com','pascal')
```
Comment (baciu-radu) [inline SQLQuery1.sql]: delete doctors and patients for  now

**Review by baciu-radu — CHANGES_REQUESTED** [review]

**Review by adrianlibenciuc — DISMISSED** [review]
Bravo

**Review by baciu-radu — DISMISSED** [review]

**Review by baciu-radu — DISMISSED** [review]

File: schema.sql [hunk 18]
```diff
@@ -0,0 +1,84 @@
...
+
+DROP TABLE IF EXISTS icn_appointment
+CREATE TABLE icn_appointment(
+	id integer NOT NULL IDENTITY PRIMARY KEY,
+	patient_id integer not null,
+	doctor_id integer not null,
+	details varchar(200) not null,
+	FOREIGN KEY (patient_id) REFERENCES icn_patient(id),
+	FOREIGN KEY (doctor_id) REFERENCES icn_doctor(id),
+	)
```
Comment (andreimladin) [inline schema.sql:72]: Indentation

File: schema.sql [hunk 19]
```diff
@@ -0,0 +1,84 @@
...
+
+DROP TABLE IF EXISTS icn_doctor_specialization
+--make primary key combo
+CREATE TABLE icn_doctor_specialization(
+	doctor_id integer not null,
+	specialization_id integer not null,
+	FOREIGN KEY (specialization_id) REFERENCES icn_specialization(id),
+	FOREIGN KEY (doctor_id) REFERENCES icn_doctor(id),
+	PRIMARY KEY (doctor_id,specialization_id),
+	)
```
Comment (andreimladin) [inline schema.sql:61]: Indentation

File: schema.sql [hunk 20]
```diff
@@ -0,0 +1,84 @@
...
+CREATE TABLE icn_user(
+	id integer not null identity primary key,
+	email varchar(80) not null,
+	pwd varchar(100) not null,
+	account_status integer not null,
+)
+
+DROP TABLE IF EXISTS icn_user_role
+CREATE TABLE icn_user_role(
+	user_id integer not null,
```
Comment (andreimladin) [inline schema.sql:21]: composed pk

File: schema.sql [hunk 21]
```diff
@@ -0,0 +1,84 @@
...
+	role_id integer not null,
+	FOREIGN KEY (user_id) REFERENCES icn_user(id),
+	FOREIGN KEY (role_id) REFERENCES icn_role(id),
+	)
+
+DROP TABLE IF EXISTS icn_patient
+CREATE TABLE icn_patient(
+	id integer NOT NULL IDENTITY PRIMARY KEY,
+	id_user integer NOT NULL,
+	cnp varchar(14) not null,
```
Comment (andreimladin) [inline schema.sql:31]: unique cnp

File: schema.sql [hunk 22]
```diff
@@ -0,0 +1,84 @@
...
+	)
+
+DROP TABLE IF EXISTS icn_doctor_specialization
+--make primary key combo
+CREATE TABLE icn_doctor_specialization(
+	doctor_id integer not null,
+	specialization_id integer not null,
+	FOREIGN KEY (specialization_id) REFERENCES icn_specialization(id),
+	FOREIGN KEY (doctor_id) REFERENCES icn_doctor(id),
+	PRIMARY KEY (doctor_id,specialization_id),
```
Comment (andreimladin) [inline schema.sql:60]: pk should be above foreign key

**Review by andreimladin — CHANGES_REQUESTED** [review]

File: schema.sql (hunk 22 above)
Comment (AlinaSastras) [reply schema.sql:60]: nu pricep

File: schema.sql (hunk 19 above)
Comment (AlinaSastras) [reply schema.sql:61]: what do you mean?

File: schema.sql (hunk 22 above)
Comment (baciu-radu) [reply schema.sql:60]: Primary key trebuie pus deasupra la Foreign key ca linie, asta am inteles eu :D

Comment (adrianlibenciuc) [reply schema.sql:60]: specialization_id integer not null,
        PRIMARY KEY (doctor_id,specialization_id),
        FOREIGN KEY (doctor_id) REFERENCES icn_doctor(id),
        FOREIGN KEY (specialization_id) REFERENCES icn_specialization(id),

**Review by andreimladin — DISMISSED** [review]

**Review by adrianlibenciuc — DISMISSED** [review]

**Review by adrianlibenciuc — DISMISSED** [review]

**Review by andreimladin — APPROVED** [review]

**Review by adrianlibenciuc — APPROVED** [review]

**Review by baciu-radu — APPROVED** [review]

**Pull request merged**